src/EditorFeatures/Core.Wpf/AsyncCompletion/LanguageServerSnippetExpanderAdapter.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServer.Client.Snippets;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.AsyncCompletion
+{
+    [Export(typeof(ILanguageServerSnippetExpander))]
+    [Shared]
+    internal sealed class LanguageServerSnippetExpanderAdapter : ILanguageServerSnippetExpander
```

Comment:
Can we add a comment about why this adapter is necessary? Its unclear to me if we want to keep this type around forever or if this is a temp workaround

Reply:
nevermind I remember Sams answer for this. This is because of layering